proto/options.proto

```diff
@@ -45,6 +45,7 @@ enum Costing {
   motorcycle = 10;
   auto_data_fix = 11;
   taxi = 12;
+  none_ = 13;
```

Comment:
so previously locate allowed you to pass in no costing which meant basically no rules applied. this was fine for the simple reach computation but the exact one requires a full costing object to do expansion. so i made a none type costing, which is only allowed for locate.

Reply:
Now that `none_` exists, can we pass around `const DynamicCost &` instead of the shared ptr?  I see several places where it's assumed the `shared_ptr` will not be null anyway.